error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/ImplicitBlockingFlux.java

```diff
@@ -0,0 +1,132 @@
+ */
+@AutoService(BugChecker.class)
+@BugPattern(
+    summary = "Accidental blocking of `Flux` with convenience method",
+    link = BUG_PATTERNS_BASE_URL + "ImplicitBlockingFlux",
+    linkType = CUSTOM,
+    severity = SUGGESTION,
+    tags = STYLE)
+public final class ImplicitBlockingFlux extends BugChecker implements MethodInvocationTreeMatcher {
+  private static final long serialVersionUID = 1L;
+  private static final Matcher<ExpressionTree> FLUX_WITH_IMPLICIT_BLOCK =
+      instanceMethod()
+          .onDescendantOf(Suppliers.typeFromString("reactor.core.publisher.Flux"))
+          .namedAnyOf("toIterable", "toStream")
+          .withNoParameters();
+  private static final Supplier<Type> STREAM = Suppliers.typeFromString("java.util.stream.Stream");
+
+  /** Instantiates a new {@link ImplicitBlockingFlux} instance. */
+  public ImplicitBlockingFlux() {}
+
+  @Override
+  public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState state) {
+    if (!FLUX_WITH_IMPLICIT_BLOCK.matches(tree, state)) {
+      return Description.NO_MATCH;
+    }
+
+    Description.Builder description = buildDescription(tree);
+
+    description.addFix(SuggestedFixes.addSuppressWarnings(state, canonicalName()));
+    if (ThirdPartyLibrary.GUAVA.isIntroductionAllowed(state)) {
+      trySuggestFix("com.google.common.collect.ImmutableList.toImmutableList", tree, state)
+          .ifPresent(description::addFix);
+    }
+    trySuggestFix("java.util.stream.Collectors.toList", tree, state).ifPresent(description::addFix);
+
+    return description.build();
+  }
+
+  private static Optional<SuggestedFix> trySuggestFix(
+      String fullyQualifiedMethodInvocation, MethodInvocationTree tree, VisitorState state) {
+    SuggestedFix.Builder fix = SuggestedFix.builder();
+    String replacement =
+        SuggestedFixes.qualifyStaticImport(fullyQualifiedMethodInvocation, fix, state);
+
+    return replaceMethodInvocationWithCollect(tree, replacement + "()", fix.build(), state);
+  }
+
+  private static Optional<SuggestedFix> replaceMethodInvocationWithCollect(
+      MethodInvocationTree tree,
+      String collectArgument,
+      SuggestedFix additionalFix,
+      VisitorState state) {
+    String collectMethodInvocation = String.format("collect(%s)", collectArgument);
+    Types types = state.getTypes();
+    String postFix =
+        types.isSubtype(ASTHelpers.getResultType(tree), types.erasure(STREAM.get(state)))
+            ? ".block().stream()"
+            : ".block()";
+    return replaceMethodInvocation(tree, collectMethodInvocation, state)
+        .map(fix -> fix.merge(additionalFix))
+        .map(fix -> fix.postfixWith(tree, postFix))
+        .map(SuggestedFix.Builder::build);
+  }
+
+  private static Optional<SuggestedFix.Builder> replaceMethodInvocation(
+      MethodInvocationTree tree, String replacement, VisitorState state) {
+    ImmutableList<ErrorProneToken> tokens =
```

Comment:
I found a way to do this without the `ErrorProneToken`! I'll push the changes 😄.